executor/common_kvm_arm64.h

```diff
@@ -16,6 +22,52 @@ struct kvm_opt {
 	uint64 val;
 };
 
+// Call KVM_SET_USER_MEMORY_REGION for the given pages.
+static void vm_set_user_memory_region(int vmfd, uint32 slot, uint32 flags, uint64 guest_phys_addr, uint64 memory_size, uint64 userspace_addr)
+{
+	struct kvm_userspace_memory_region memreg;
+	memreg.slot = slot;
+	memreg.flags = flags;
+	memreg.guest_phys_addr = guest_phys_addr;
+	memreg.memory_size = memory_size;
+	memreg.userspace_addr = userspace_addr;
+	ioctl(vmfd, KVM_SET_USER_MEMORY_REGION, &memreg);
+}
+
+// Set the value of the specified register.
+static void vcpu_set_reg(int vcpu_fd, uint64 id, uint64 val)
+{
+	struct kvm_one_reg reg = {.id = id, .addr = (uint64)&val};
+	ioctl(vcpu_fd, KVM_SET_ONE_REG, &reg);
```

Comment:
I see there are dozens of special registers setting which invokes custom kernel code:
https://elixir.bootlin.com/linux/v6.10/source/arch/arm64/kvm/guest.c#L793

I don't see any of these in our descriptions (KVM_REG_ARM_CORE, KVM_REG_ARM_TIMER_CTL, [KVM_REG_ARM_STD_BMAP](https://elixir.bootlin.com/linux/v6.10/C/ident/KVM_REG_ARM_STD_BMAP), etc).

Adding these may give some low hanging coverage.

Reply:
Good idea, thank you!
I'll send those separately at some point.